### PR TITLE
kata-runtime: Return correct kata-env on ppc64le

### PR DESCRIPTION
**Pull request by nitkon**
The contents of /proc/cpuinfo were
trimmed and hence the "model" field could
not be parsed despite being a field in
/proc/cpuinfo. Fix this issue.

Fixes: #1089

Signed-off-by: Nitesh Konkar niteshkonkar@in.ibm.com

### CODE REVIEW

File: cli/kata-check.go [hunk 1]
```diff
@@ -76,14 +76,11 @@ func getCPUInfo(cpuInfoFile string) (string, error) {
 		return "", err
 	}
 
-	cpus := strings.SplitAfter(text, "\n\n")
-
-	trimmed := strings.TrimSpace(cpus[0])
-	if trimmed == "" {
+	if len(strings.TrimSpace(text)) == 0 {
```
Comment (jodh-intel) [inline cli/kata-check.go]: This function is now returning *all* details for *all* cpus fwics. This is to accommodate PPC64le which has an unusual `/proc/cpuinfo` - it contains an extra paragraph at the end which includes the `model` details:

- https://github.com/kata-containers/runtime/blob/master/cli/kata-check_data_ppc64le_test.go#L111

I guess there is no harm *per se* in returning all the duplicated information but it still feels a bit odd to me. Assuming this works as expected on other platforms, please can you update the comment for the function as it is now incorrect as it states:

> // getCPUInfo returns details of **the first CPU** read from the specified cpuinfo file

Hi @Pennyzct and @alicefr - Happy 2019! :smile: Please could you take a look from the ARM64 and os390x perspectives please?

**Comment by jodh-intel** [comment]
I forgot to say, how about adding a ppc64le-specific unit-test which either actually runs a real `/proc/cpuinfo` check to ensure the expected `model` value is returned by the function, or which creates a fake `/proc/cpuinfo` containing `model` and again checks the returned value?

File: cli/kata-check.go (hunk 1 above)
Comment (Pennyzct) [reply cli/kata-check.go]: Hi~ @jodh-intel, Happy 2019! :) ARM64 share the same structure as amd64, all duplicated info. Do we need to make this func arch-dependent??

**Comment by grahamwhaley** [comment]
You may need to make the `cpuinfo` funcs arch dependant - iirc there is no standard for the info that comes out of `/proc/cpuinfo` across architectures (btdt)

File: cli/kata-check.go (hunk 1 above)
Comment (jodh-intel) [reply cli/kata-check.go]: Yep - I agree with both you and @grahamwhaley @Pennyzct - it would be much clearer to have arch-specific functions for this.

**Comment by jodh-intel** [comment]
@nitkon - I'll add the dnm label for now so you can add a `getPPC64leCPUDetails()` function in a similar manner to `getS390xCPUDetails()`...

**Comment by nitkon** [comment]
ping for review ...

File: cli/kata-check_ppc64le.go [hunk 2]
```diff
@@ -97,12 +97,62 @@ func archKernelParamHandler(onVMM bool, fields logrus.Fields, msg string) bool {
...
+			}
+		}
+	}
+
+	if archCPUVendorField != "" && vendor == "" {
+		return "", "", fmt.Errorf("cannot find vendor field in file %v", procCPUInfo)
+	}
+
+	// model name is optional
+	if archCPUModelField != "" && model == "" {
```
Comment (caoruidong) [inline cli/kata-check_ppc64le.go]: Is it optional?

Comment (nitkon) [reply cli/kata-check_ppc64le.go]: Nope. Thanks! Fixed.

**Comment by jodh-intel** [comment]
Thanks @nitkon.

lgtm

**Comment by nitkon** [comment]
All green. Good to merge. @jodh-intel @caoruidong

**Pull request merged**